internal/tlscert/tls_cert_validator.go

```diff
@@ -19,25 +19,48 @@ import (
 )
 
 var (
-	ErrCertDecodeFailed = errors.New("failed to decode PEM block containing cert")
+	ErrCertDecodeFailed = errors.New("failed to decode PEM block containing certificate")
 	ErrCertParseFailed  = errors.New("failed to parse certificate")
 
 	ErrKeyDecodeFailed = errors.New("failed to decode PEM block containing private key")
 	ErrKeyParseFailed  = errors.New("failed to parse private key")
 
+	ErrCADecodeFailed = errors.New("failed to decode PEM block containing CA certificate")
+	ErrCAParseFailed  = errors.New("failed to parse CA certificate")
+
 	ErrValueResolveFailed = errors.New("failed to resolve value")
 
 	ErrInvalidCertificateKeyPair = errors.New("certificate and private key do not match")
+
+	ErrCertIsNotCA = errors.New("not a CA certificate")
 )
 
+type TLSBundle struct {
+	Cert *telemetryv1alpha1.ValueType
+	Key  *telemetryv1alpha1.ValueType
+	CA   *telemetryv1alpha1.ValueType
+}
+
 const twoWeeks = time.Hour * 24 * 7 * 2
 
+const (
+	CertExpiredErrorMessage  = "TLS certificate expired on %s"
+	CertAboutToExpireMessage = "TLS certificate is about to expire, configured certificate is valid until %s"
+	CaExpiredErrorMessage    = "TLS CA expired on %s"
```

Comment:
Let's use "CA certificate" in the message. A CA as a (legal) entity will usually not expire. Only a certificate that it uses.